clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -222,6 +225,7 @@ public ConsumerCoordinator(GroupRebalanceConfig rebalanceConfig,
         }
 
         this.metadata.requestUpdate();
+        this.partitionOffsetsCache = new ConcurrentHashMap<>();
```

Comment:
In consumer, there are 2 threads currently, 1 for heartbeat, 1 for main consumer thread. From my understanding, the heartbeat thread won't do offsets committing. If so, using HashMap should be fine. Could you help confirm it? Check `AbstractCoordinator#HeartbeatThread`.

Reply:
I'll double check this and move to using a simple HashMap if doable, thanks for calling it out!